pkg/proxy/apis/kubeproxyconfig/validation/validation_test.go

```diff
@@ -340,22 +340,26 @@ func TestValidateKubeProxyIPTablesConfiguration(t *testing.T) {
 	}
 
 	for _, errorCase := range errorCases {
-		if errs := validateKubeProxyIPTablesConfiguration(errorCase.config, newPath.Child("KubeProxyIPTablesConfiguration")); len(errs) == 0 {
-			t.Errorf("expected failure for %s", errorCase.msg)
-		} else if !strings.Contains(errs[0].Error(), errorCase.msg) {
-			t.Errorf("unexpected error: %v, expected: %s", errs[0], errorCase.msg)
-		}
+		t.Run(errorCase.msg, func(t *testing.T) {
+			if errs := validateKubeProxyIPTablesConfiguration(errorCase.config, newPath.Child("KubeProxyIPTablesConfiguration")); len(errs) == 0 {
+				t.Errorf("expected failure for %s", errorCase.msg)
```

Comment:
t.Fatalf("expected failure")

Can use Fatal rather than Error since Fatal will now only exit the subtest.
And the error message no longer needs to list the test name since that will already be printed as part of the failing subtest name.

(Likewise just below, and likewise in several other tests which now have redundant names and/or index numbers in their error messages. Also, you don't need the "else" on the line below this any more now either.)